src/Setup/UpdateVersions.purs

```diff
@@ -70,29 +73,74 @@ fetchLatestReleaseVersion tool = Tool.repository tool # case tool of
   -- TODO: These functions really ought to be in ExceptT to avoid all the
   -- nested branches.
   fetchFromGitHubReleases repo = recover do
-    let url = "https://api.github.com/repos/" <> repo.owner <> "/" <> repo.name <> "/releases/latest"
-
-    AX.get RF.json url >>= case _ of
-      Left err -> do
-        throwError (error $ AX.printError err)
-
-      Right { body } -> case (_ .: "tag_name") =<< decodeJson body of
-        Left e -> do
-          throwError $ error $ fold
-            [ "Failed to decode GitHub response. This is most likely due to a timeout.\n\n"
-            , printJsonDecodeError e
-            , stringify body
-            ]
-
-        Right tagStr -> do
-          let tag = fromMaybe tagStr (String.stripPrefix (String.Pattern "v") tagStr)
-          case Version.parseVersion tag of
-            Left e ->
-              throwError $ error $ fold
-                [ "Failed to decode tag from GitHub response: ", parseErrorMessage e ]
-
-            Right v ->
-              pure v
+    page <- liftEffect (Ref.new 1)
+    untilJust do
+      versions <- liftEffect (Ref.read page) >>= toolVersions repo
+      when (isNothing versions)
+        $ throwError
+        $ error
+            "Could not find version that is not a pre-release version"
+      let
+        version =
+          versions
+            # unsafePartial fromJust
```

Comment:
I know I probably should be using a `case` instead of `unsafePartial` here, but I kinda got sick of `case` because of the `case` hell below. If you don't like this I can make it a `case` instead. I do plan though on making a PR that makes these functions use `ExceptT`.